editor/store/actions.js

```diff
@@ -387,14 +388,34 @@ export function mergeBlocks( blockAUid, blockBUid ) {
 }
 
 /**
- * Returns an action object used in signalling that the post should autosave.
- *
- * @return {Object} Action object.
+ * Action creator triggering the autosave behavior
  */
-export function autosave() {
-	return {
-		type: 'AUTOSAVE',
-	};
+export function* autosave() {
+	const {
+		isEditedPostSaveable,
+		isEditedPostDirty,
+		isEditedPostNew,
+		isCurrentPostPublished,
+	} = select( 'core/editor' );
+
+	if ( ! isEditedPostSaveable() ) {
```

Comment:
To access the state we're calling `wp.data.select` directly. Another approach would be to inject a `select` helper somehow into the generator:


```js
export function* autosave( select ) {
   const isSaveable = select( 'core/editor' ).isEditedPostSaveable(); // or just `select.isEditedPostSaveable()`
}
```

The downside of this approach is that it adds an extra argument to the action creator which is not ideal.
Maybe it's fine to rely on the global and mock the global in the tests. Thoughts?